### PR TITLE
Allow skipping parts of storylines

### LINKED ISSUE #47: Mission skip menu
https://github.com/EndlessSkyCommunity/world-forge/issues/28 world forge had this suggestion first.

Since I've been sitting on this consideration for a long while, I may well have a record of it. Skipping missions should be made to work independently of each other, but chained skips should only display in order and not alongside each other (to avoid situations like having the Wanderer story done but not FW).

Skips
- Sheragi archaeology 1/2 (to `Rim Archaeology 6`)
- FW story 1/2 (to `FW Defend New Tibet`)
   - FW story 2/2 (recon) (from `FW Reconciliation 1`)
   - FW story 2/2 (check) (from `FWC Scouting 1`)
      - Wanderer story 1/2 (to `Wanderers Ap'arak 3`)
         - Wanderer story 2/2 (from `Wanderers: Pre-Colony`)
      - Sheragi archaeology 2/2 (from `Sheragi Archaeology: The Box 1`)
- Remnant story 1/2 (to `Remnant: Technology Available`)
   - Remnant story 2/2 (from `Remnant: Learn Sign 1`)

### PR DESCRIPTION
**Pull request by TheGiraffe3**
Fixes #47.

Also mixed a rename in.

### CODE REVIEW

**Review by DarcyManoel — COMMENTED** [review]
As I understand this, the missions will mostly offer one after the other, but it seems like it would require taking off and landing until you get to the point you wanted. It may be a better experience to have the one mission with a conversation where the choices have `to display` options that are conditionally available so that players can get to where they want from one conversation.

Also are these jobs meant to be offered anywhere? Wouldn't it make more sense to be offered within the Omnis systems?

**Comment by TheGiraffe3** [comment]
> Also are these jobs meant to be offered anywhere?

Ummm... no, actually. I forgot the `source` tag.

I'll do the single mission approach.

**Review by DarcyManoel — APPROVED** [review]
Assuming you've tested thoroughly, looks good.

File: data/missions/mod settings.txt [hunk 1]
```diff
@@ -577,6 +579,415 @@ mission "Omnis: mod settings"
...
+				event "Deep memorial completed"
+				event "geminus rebuilt"
+				event "martini recovers"
+			`The Free Worlds plot has been marked as complete.`
+				goto 1700
+			label 1711
+			action
+				set "FW Rand 1B: done"
+				set "main plot completed"
+				set "omnis plugin: skipped free worlds halfway"
```
Comment (TheGiraffe3) [inline data/missions/mod settings.txt:715]:
```suggestion
				set "omnis: skipped free worlds halfway"
```

File: data/missions/mod settings.txt [hunk 2]
```diff
@@ -577,6 +579,415 @@ mission "Omnis: mod settings"
...
+				event "bloodsea independent"
+				event "albatross joins free worlds"
+				event "fw expanded and cut"
+			`You have successfully skipped ahead in the Free Worlds storyline until the branch between Checkmate and Reconciliation. You can either go to Wayfarer and land to continue the Free Worlds storyline or skip further ahead.`
+				goto 1700
+			label 1712
+			action
+				set "FWC Pug 6: done"
+				set "main plot completed"
+				set "omnis plugin: skipped free worlds checkmate"
```
Comment (TheGiraffe3) [inline data/missions/mod settings.txt:768]:
```suggestion
				set "omnis: skipped free worlds checkmate"
```

File: data/missions/mod settings.txt [hunk 3]
```diff
@@ -577,6 +579,415 @@ mission "Omnis: mod settings"
...
+				event "Deep memorial completed"
+				event "geminus rebuilt"
+				event "martini recovers"
+			`You have successfully skipped ahead in the Free Worlds Checkmate storyline until the last mission. Land on Pugglemug to get the final mission in Free Worlds Checkmate.`
+				goto 1700
+			label 1713
+			action
+				set "FW Syndicate Extremists 1B: done"
+				set "main plot completed"
+				set "omnis plugin: skipped free worlds reconciliation"
```
Comment (TheGiraffe3) [inline data/missions/mod settings.txt:846]:
```suggestion
				set "omnis: skipped free worlds reconciliation"
```

File: data/missions/mod settings.txt [hunk 4]
```diff
@@ -577,6 +579,415 @@ mission "Omnis: mod settings"
...
+						has "Wanderers Rek 3: done"
+						not "Wanderers: Sestor: Scan Drones: done"
+				`	(Go back to Save Settings.)`
+					goto 1000
+				`	(Go back to Settings.)`
+					goto 0000
+			label 1710
+			action
+				set "main plot completed"
+				set "omnis plugin: skipped free worlds"
```
Comment (TheGiraffe3) [inline data/missions/mod settings.txt:634]:
```suggestion
				set "omnis: skipped free worlds"
```

**Pull request merged**